packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -225,8 +228,21 @@ const DropdownMenuRootContent = React.forwardRef((props, forwardedRef) => {
       portalled={portalled}
       onCloseAutoFocus={composeEventHandlers(props.onCloseAutoFocus, (event) => {
         event.preventDefault();
-        context.triggerRef.current?.focus();
+        if (!preventCloseAutoFocusRef.current) context.triggerRef.current?.focus();
+      })}
+      onEscapeKeyDown={composeEventHandlers(props.onEscapeKeyDown, () => {
+        preventCloseAutoFocusRef.current = false;
       })}
```

Comment:
Do we need `{ checkForDefaultPrevented: false }` here? In fact, do we need it on all of the handlers that set the handlers in this PR that set `isPointerDownOutsideRef` or `preventCloseAutoFocusRef` ?

Reply:
I thought we said we would re-assess all this separately?

Reply:
Honestly I wasn't on which way to go here, we did say we'd assess it separately so I just carried over the existing stance from the pointer handlers in `Popover` and `Menu`.

Reply:
Oh that's true, i'll revert it on `onEscapeKeyDown` just to keep it inline with how it was for now.